doc/storage.md

```diff
@@ -0,0 +1,170 @@
+storage proposal will not automatically reuse existing LVM structures.
+
+To reuse existing volume groups the user must explicitly specify that. See the section "future
+features".
+
+## Agama and AutoYaST
+
+The relationship between the Agama storage schema and the old AutoYaST format is described
+at a [separate document](./autoyast_storage.md).
+
+## Calculating the omitted size of a file system
+
+If the size is omitted for a new device that directly contain a `filesystem` entry with a mount
+point, Agama will then use the settings of the product to set the size limits. From a more
+technical point of view, that translates into the following:
+
+ - If the mount path corresponds to a volume supporting `auto_size`, that feature will be used.
+ - If it corresponds to a volume without `auto_size`, the min and max sizes of the volumes will be
+   used.
+ - If there is no volume for that mount path, the sizes of the default volume will be used.
+ - If the product does not specify a default volume, the behavior is still not defined (there are
+   several reasonable options).
+
+## Schema sections under discussion
+
+This section summarizes several aspects of the Agama storage schema that have been considered
+but not implemented so far.
+
+### Specifying the Size of a Device
+
+The current schema makes it possible to specify the sizes as a fixed value or as a range. But a
+a parseable string like "40 GiB" may not be the only option to represent a size or a range limit.
+The following two possibilities are also under consideration.
+
+ - `{ "gib": 40 }`
+ - `{ "value": 40, "units": "gib" }`
+
+### Searching Existing Devices
+
+Strings may be used as value for `search` to locate a device by its name or to search all existing
+devices using "\*". But strings may be useful in other situations.
+
+For example, "next" (or any similar term) could be used to represent the default search for drives
+(which is something like `{ "sort": { "property": "name" }, "max": 1, "ifNotFound": "error" }`.
+
+If a simple string like "next" could be used to specify the standard search entry for drives, it
+would make sense to simply make `search` mandatory for all drives instead of assuming a default one.
+
+Another possible improvement for that string-based format would be supporting regular expressions.
+That would make it possible to use searchers like this.
+
+```json
+{ "search": ".*" }
+```
+
+But regular expressions would not play well with libstorage-ng. Since not all device names are
+stored in the devicegraph, it is is necessary to use functions like `find_by_any_name` in order to
+perform an exhaustive search by name.
+
+Another aspect under discussion is the format to specify conditions. Instead of the format described
+above, it would be possible to use the key as name of the property, resulting in something like this.
+
+```json
+{
+    "search": {
+        "condition": { "sizeGib": 1, "operator": "greater" }
```

Comment:
Another option (and my favourite one) is:

```json
"condition": { "size": { "greater": "1 GiB" } } 
```

Reply:
Added to the document.